common/djangoapps/student/tests/test_views.py

```diff
@@ -380,6 +381,26 @@ def test_unfulfilled_entitlement(self, mock_course_overview, mock_pseudo_session
         self.assertIn('<div class="course-entitlement-selection-container ">', response.content)
         self.assertIn('Related Programs:', response.content)
 
+        # If an entitlement has already been redeemed by the user for a course run, do not let the run be selectable
+        enrollment = CourseEnrollmentFactory(user=self.user, course_id=unicode(mock_course_overview.return_value.id))
+        CourseEntitlementFactory.create(
+            user=self.user, course_uuid=program['courses'][0]['uuid'], enrollment_course_run=enrollment)
```

Comment:
You might not need this part of the test anymore, right? (you could just create a bare enrollment)  But I can see an argument for keeping it, as it is the typical path.  But excess test code always makes me wary.

Reply:
Right, I don't need the second entitlement anymore just the enrollment, but I think I'll keep it this way just because it tends to follow the user workflow that we would expect.